File: app/src/main/java/com/tianchuang/ihome_b/mvp/BasePresenterImpl.java
package com.tianchuang.ihome_b.mvp;

import android.content.Context;

/**
 * Created by Abyss on 2017/4/1.
 * description:p的通用实现
 */

public class BasePresenterImpl<V extends BaseView> implements BasePresenter<V> {
    protected V mView;

    @Override
    public void attachView(V view) {
        this.mView = view;
    }

    @Override
    public void detachView() {
        this.mView = null;
    }

    public Context getContext() {//最好在获取资源时使用
        return mView.getContext();
    }

}

File: app/src/main/java/com/tianchuang/ihome_b/mvp/BasePresenterImpl.kt
package com.tianchuang.ihome_b.mvp

import android.content.Context

/**
 * Created by Abyss on 2017/4/1.
 * description:p的通用实现
 */

open class BasePresenterImpl<V : BaseView> : BasePresenter<V> {
     protected var mView: V? = null

    override fun attachView(view: V) {
        this.mView = view
    }

    override fun detachView() {
        this.mView = null
    }

    //最好在获取资源时使用
    val context: Context
        get() = mView!!.getContext()

}
